### PR TITLE
Reset Password Frontend

### PR DESCRIPTION
**Pull request by gblake31**
Pretty self-explanatory, you can watch the video for details. That annoying popup is a firefox thing, I promise

-Reset password on the profile page if you're logged in, on the login page if you aren't.
-Now you can submit the login/register with the enter key. 
-Got rid of Google/Github buttons.
-It lets you change it to anything, including the previous password you had.

https://github.com/ofast-team/frontend/assets/51721851/e515915d-5dd1-48f2-b5bc-08c99e44ea72

Okay, I'm no longer logging out the user, just telling them I sent an email to reset their password. They cannot spam this button, they can only click it once each time they refresh the page.
image

### CODE REVIEW

**Review by tylerm390 — COMMENTED** [review]
Looks good to me but might be good to have a frontend person review as well.

**Review by sd0623 — APPROVED** [review]
LGTM

File: public/lessons/runtime-analysis/problem-solving.mdx [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 A helpful rule-of-thumb in competitive programming is that computers can perform around $10^8$ operations in a second.
 You'll gain a better intuition for how much you can push this limit as you solve more problems.
 
 On O(fast), problems can be submitted in C++, Java, and Python. Of the three languages, C++ runs the fastest.
 Java runs a bit slower than C++, and Python runs a bit slower than Java. Most problems on O(fast) will pass in all 3 languages as
-long as the Big-O of the solution is small enough. However, programming contest on websites like Codeforces and Kattis won't be as forgiving.
-It's important to consider the performance of a programming language when deciding to learn it. 
-
+long as the Big-O of the solution is small enough. However, collegiate programming contests like the ICPC won't be as forgiving.
+We (the O(fast) team) highly recommend learning C++ for competitive programming. C++ can be difficult to troubleshoot sometimes, but its syntax is concise and it's performance is unmatched.
```
Comment (AsirAAlam) [inline public/lessons/runtime-analysis/problem-solving.mdx:11]: *its* performance

File: public/lessons/runtime-analysis/problem-solving.mdx [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
 
 A helpful rule-of-thumb in competitive programming is that computers can perform around $10^8$ operations in a second.
 You'll gain a better intuition for how much you can push this limit as you solve more problems.
 
 On O(fast), problems can be submitted in C++, Java, and Python. Of the three languages, C++ runs the fastest.
 Java runs a bit slower than C++, and Python runs a bit slower than Java. Most problems on O(fast) will pass in all 3 languages as
-long as the Big-O of the solution is small enough. However, programming contest on websites like Codeforces and Kattis won't be as forgiving.
-It's important to consider the performance of a programming language when deciding to learn it. 
-
+long as the Big-O of the solution is small enough. However, collegiate programming contests like the ICPC won't be as forgiving.
```
Comment (AsirAAlam) [inline public/lessons/runtime-analysis/problem-solving.mdx:10]: Maybe it's just me, but "the Big-O of the solution" sounds somehow... improper. 
Do you think we can instead say something like "the time complexity of the solution is fast enough"?

File: src/components/ProfileCard.tsx [hunk 3]
```diff
@@ -268,9 +298,17 @@ export default function ProfileCard({
           </Grid>
         </div>
       </Stack>
-      <LoginButton sx={{ fontSize: '20px', marginTop: '20px', width: '100%' }}>
-        Change Password
+      <LoginButton
```
Comment (AsirAAlam) [inline src/components/ProfileCard.tsx:301]: lol we should rename this button at some point

File: src/pages/LoginPage.tsx [hunk 4]
```diff
@@ -138,6 +132,13 @@ export default function LoginPage() {
       })
   }
 
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      console.log('enter pressed')
```
Comment (AsirAAlam) [inline src/pages/LoginPage.tsx:137]: remove log

File: src/pages/LoginPage.tsx [hunk 5]
```diff
@@ -155,32 +156,34 @@ export default function LoginPage() {
...
-              fontSize={'18px'}
-              color={'red'}
-              visibility={hasInvalidCredentials ? 'visible' : 'hidden'}
-            >
-              Invalid Credentials, please try again.
-            </Typography>
+            <PasswordField
+              setter={setPassword}
+              onKeyPress={handleKeyPress}
+            ></PasswordField>
```
Comment (AsirAAlam) [inline src/pages/LoginPage.tsx:162]: nit: we can self close the tag. and the textfield above this

```
<PasswordField
  setter={setPassword}
  onKeyPress={handleKeyPress}
/>
```

File: src/pages/RegisterPage.tsx [hunk 6]
```diff
@@ -107,7 +71,10 @@ export default function RegisterPage() {
               variant="standard"
               onChange={(e) => setEmail(e.target.value)}
             ></TextField>
-            <PasswordField setter={setPassword}></PasswordField>
+            <PasswordField
+              setter={setPassword}
+              onKeyPress={handleKeyPress}
+            ></PasswordField>
```
Comment (AsirAAlam) [inline src/pages/RegisterPage.tsx:77]: nit: self close tag

**Review by AsirAAlam — COMMENTED** [review]
Sorry for late review!

**Pull request merged**